.github/workflows/workflow.yml

```diff
@@ -78,3 +80,23 @@ jobs:
         with:
           name: artifacts
           path: otel-dotnet-autoinstrumentation-macos.tar
+
+  create_release:
+    name: Create release
+    runs-on: ubuntu-20.04
+    if: startsWith(github.ref, 'refs/tags/') # only trigger if the current event is for a tag push
+    needs: [alpine_build, linux_build, windows_build, macos_build]
+    steps:
+      - uses: actions/download-artifact@v2
+        with:
+          path: .
+      - name: Create Release
+        uses: softprops/action-gh-release@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
Request only necessary permissions of the GITHUB_TOKEN on the job level. See: https://docs.github.com/en/actions/reference/authentication-in-a-workflow#permissions-for-the-github_token

I guess it will be:
```
permissions:
  contents: write
```

Please test it on a fork